gadget/install/install.go

```diff
@@ -222,6 +231,24 @@ func Run(model gadget.Model, gadgetRoot, kernelRoot, device string, options Opti
 		}
 	}
 
+	// after we have created all partitions, build up the mapping of volumes
+	// to disk device traits and save it to disk for later usage
+	optsPerVol := map[string]*gadget.DiskVolumeValidationOptions{
+		// this assumes that the encrypted partitions above are always only on the
+		// system-boot volume, this assumption may change
+		laidOutBootVol.Name: {
+			ExpectedStructureEncryption: partsEncrypted,
+		},
+	}
+	allVolTraits, err := gadget.AllDiskVolumeDeviceTraits(allLaidOutVols, optsPerVol)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := gadget.SaveDiskVolumesDeviceTraits(dirs.SnapDeviceDirUnder(boot.InstallHostWritableDir), allVolTraits); err != nil {
```

Comment:
would we have a reason to use this information from recovery? another install and factory-reset would both capture this again as well ? I'm basically wondering if  it's necessary/useful to save this to ubuntu-save, or it doesn't matter

Reply:
Right now, we would not have a reason to use it from recovery because it's only used for gadget asset updates, which right now cannot be performed from recovery by design since we don't allow refreshes to gadget snap in recovery. 

However, we could potentially some day enable some sort of gadget asset update routine to be runnable from recovery in the case where something goes wrong during a normal update, maybe we transition to recover and reinstall the assets as they were before the update to be able to get back to run mode, but considering that sort of feature is far off in the future, I think we are okay to only put this on data for now and when we want to be able to use it from save then we start storing a copy in save. 

Does that sound reasonable?

Reply:
to not end up to devices that can't support the feature without extra copying, it might be cheap/easy enough to store both in data and save (if it exists), and read only from data for now?

Reply:
sure we can store it in save too

Reply:
done now